pkg/deploy/api/v1/types.go

```diff
@@ -28,12 +28,17 @@ const (
 type DeploymentStrategy struct {
 	// Type is the name of a deployment strategy.
 	Type DeploymentStrategyType `json:"type,omitempty" description:"the name of a deployment strategy"`
+
 	// CustomParams are the input to the Custom deployment strategy.
 	CustomParams *CustomDeploymentStrategyParams `json:"customParams,omitempty" description:"input to the Custom deployment strategy"`
 	// RecreateParams are the input to the Recreate deployment strategy.
 	RecreateParams *RecreateDeploymentStrategyParams `json:"recreateParams,omitempty" description:"input to the Recreate deployment strategy"`
 	// RollingParams are the input to the Rolling deployment strategy.
 	RollingParams *RollingDeploymentStrategyParams `json:"rollingParams,omitempty" description:"input to the Rolling deployment strategy"`
+
+	// After is an optional set of actions to take after a deployment has completed.
+	After *AfterDeploymentActions `json:"after" description:"actions to take after a deployment has completed"`
```

Comment:
What about:

```go
type DeploymentStrategy struct {
    // ..
    After []AfterDeploymentAction
    // ..
}

type AfterDeploymentAction struct {
    TagImage *TagImageAction
    // New actions can go here
}

type TagImageAction struct {
    // ...
}
```

Not only does it look much more obvious what actions you have when there is more than one action type defined, but you get ordered actions for free (if we want to support that on the processing side).